File: app/schemas/workspace.py
import uuid
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class WorkspaceBase(BaseModel):
    client_id: uuid.UUID
    name: str
    status: str = "active"


class WorkspaceCreate(WorkspaceBase):
    pass


class WorkspaceUpdate(BaseModel):
    name: Optional[str] = None
    status: Optional[str] = None


class WorkspaceOut(WorkspaceBase):
    id: uuid.UUID
    created_by: Optional[uuid.UUID] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/workspace.ts
// ============================================================
// Company OS — Workspace Types
// ============================================================

export interface Workspace {
  id: string;
  client_id: string;
  name: string;
  status: string;
  drive_folder_id?: string;
  drive_url?: string;
  created_by?: string;
  created_at: string;
  updated_at: string;
}

export interface WorkspaceCreate {
  client_id: string;
  name: string;
  status?: string;
  drive_folder_id?: string;
}

export interface WorkspaceUpdate {
  name?: string;
  status?: string;
  drive_folder_id?: string;
}
